packages/contracts/scripts/deploy/manager/DiamondInAll.ts

```diff
@@ -0,0 +1,195 @@
+    owner: string,
+    init: string,
+    initCalldata: string
+}
+
+export type ForgeArguments = {
+    name: string,
+    network: string,
+    rpcUrl: string,
+    privateKey: string,
+    contractInstance: string,
+    constructorArguments?: [DiamondArgs, cutType[]],
+    etherscanApiKey?: string,
+}
+
+const create = async (args: ForgeArguments): Promise<{ result: DeploymentResult | undefined, stderr: string }> => {
+    let flattenConstructorArgs = `\"(`;
+    let prepareCmd: string;
+    if (args.constructorArguments) {
+        let diamondArgs = args.constructorArguments[0];
+        let diamondCut = args.constructorArguments[1];
+
+        flattenConstructorArgs += `${diamondArgs.owner},${diamondArgs.init},${diamondArgs.initCalldata})\" \"[`;
+
+        for (let i = 0; i < diamondCut.length; i++) {
+            flattenConstructorArgs += `(${diamondCut[i].facetAddress},${diamondCut[i].action},[`;
+
+            let functionSelectors = diamondCut[i].functionSelectors;
+            for (let j = 0; j < functionSelectors.length; j++) {
+                flattenConstructorArgs += `${functionSelectors[j]}`;
+                if (j !== functionSelectors.length - 1) {
+                    flattenConstructorArgs += `,`;
+                }
+            }
+            flattenConstructorArgs += `])`;
+            if (i !== diamondCut.length - 1) {
+                flattenConstructorArgs += `,`;
+            }
+        }
+        flattenConstructorArgs += `]\"`;
+        prepareCmd = `forge create --json --rpc-url ${args.rpcUrl} --private-key ${args.privateKey} ${args.contractInstance} --constructor-args ${flattenConstructorArgs}`;
+    } else {
+        prepareCmd = `forge create --json --rpc-url ${args.rpcUrl} --private-key ${args.privateKey} ${args.contractInstance}`;
+    }
+
+    const chainId = Networks[args.network] ?? undefined;
+    if (!chainId) {
+        throw new Error(`Unsupported network: ${args.network}.`);
+    }
+
+    let executeCmd: string;
+    if (args.etherscanApiKey) {
+        executeCmd = `${prepareCmd} --etherscan-api-key ${args.etherscanApiKey} --verify`
+    } else {
+        executeCmd = prepareCmd
+    }
+    let stdout;
+    let stderr: string;
+    let result: DeploymentResult | undefined;
+
+    try {
+        const { stdout: _stdout, stderr: _stderr } = await execute(executeCmd)
+        stdout = _stdout;
+        stderr = _stderr;
+    } catch (err: any) {
+        console.log(err);
+        stdout = err?.stdout;
+        stderr = err?.stderr;
+    }
+    if (stdout) {
+        const regex = /{(?:[^{}]*|(R))*}/g;
+        const found = stdout.match(regex);
+        if (found && JSON.parse(found[0])?.deployedTo) {
+            const { deployedTo, deployer, transactionHash } = JSON.parse(found[0]);
+            result = { deployedTo, deployer, transactionHash };
+            console.log(`Deployed ${args.name} contract successfully. Address: ${deployedTo}`);
+        }
+    }
+    return { result, stderr }
+}
```

Comment:
> Why couldn't you use `packages/contracts/scripts/deploy/create.ts` instead of making a new one which is almost same?

As you can see, the constructor argument in diamond is a special case, including objects and array of objects. So didn't want to mix them with the original simple cases
Hope to lend your hand on this